src/rgen/Microsoft.Macios.Generator/Extensions/TypeSymbolExtensions.cs

```diff
@@ -0,0 +1,28 @@
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Macios.Generator.Extensions;
+
+static class TypeSymbolExtensions {
+
+	public static Dictionary<string, AttributeData> GetAttributeData (this ISymbol symbol)
+	{
+		var boundAttributes = symbol.GetAttributes ();
+		if (boundAttributes.Length == 0) {
+			// return an empty dictionary if there are no attributes
+			return new ();
+		}
+
+		var attributes = new Dictionary<string, AttributeData> ();
+		foreach (var attributeData in boundAttributes) {
+			var attrName = attributeData.AttributeClass?.ToDisplayString ();
+			if (string.IsNullOrEmpty (attrName))
+				continue;
+			if (!attributes.TryAdd (attrName, attributeData)) {
+				// TODO: diagnostics
```

Comment:
I assume that the name and signature of this method will change to a `TryGet` pattern?

Reply:
I think I need to do better with this method, there can be duplicated attributes, with diff attribute data. I'll update it accordingly.